ui/src/components/CanvasContextMenu.tsx

```diff
@@ -0,0 +1,50 @@
+import { useStore } from "zustand";
+import { RepoContext } from "../lib/store";
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import MenuList from '@mui/material/MenuList';
+import MenuItem from '@mui/material/MenuItem';
+import React, { useContext } from 'react';
+import CodeIcon from '@mui/icons-material/Code';
+import PostAddIcon from '@mui/icons-material/PostAdd';
+import FormatListNumberedIcon from '@mui/icons-material/FormatListNumbered';
+
+const paneMenuStyle = (left, top) => {
+    return ({
+        left: `${left}px`,
+        top: `${top}px`,
+        zIndex: 100,
+        position: 'absolute',
+        boxShadow: '0px 1px 8px 0px rgba(0, 0, 0, 0.1)',
+        // width: '200px',
+        backgroundColor: '#fff',
+        borderRadius: '5px',
+        boxSizing: 'border-box',
+    }) as React.CSSProperties;
+};
+
+
+export function CanvasContextMenu(props) {
+    const store = useContext(RepoContext);
+    if (!store) throw new Error("Missing BearContext.Provider in the tree");
+    const showLineNumbers = useStore(store, (state) => state.showLineNumbers);
+    const flipShowLineNumbers = useStore(store, (state) => state.flipShowLineNumbers);
+    return (
+        <div style={paneMenuStyle(props.x, props.y)}>
+            <MenuList className='paneContextMenu'>
```

Comment:
Let's don't use CSS names because I would like the styles to be defined close to the component/script. Let's inline it, e.g.,

```
<MenuList sx={{display: "flex", p: "5px", ...}}>
    <MenuItem sx={{mr: "10px"}}> .. </MenuItem>
</MenuList>
```

Follow the styles here: https://mui.com/material-ui/react-menu/